src/main/java/org/openstreetmap/atlas/geography/atlas/pbf/slicing/CountrySlicingProcessor.java

```diff
@@ -877,7 +878,12 @@ private void processWays()
             way.getTags().add(new Tag(ISOCountryTag.KEY, ISOCountryTag.COUNTRY_MISSING));
             return Optional.empty();
         }
-        else if (slices.size() == 1 || CountryBoundaryMap.isSameCountry(slices))
+        else if (slices.size() == 1 || CountryBoundaryMap.isSameCountry(slices)
+                // In case a way is "force-sliced", when it shoots into international waters, the
+                // slice number can be >1 and the country might be the same. In that case, we still
+                // want to include both slices always. This last check is to skip the coming block
+                // in that case.
+                && !this.boundaryMap.shouldForceSlicing(Taggable.with(way.getTags())))
```

Comment:
I think we need to apply this to the new Raw Atlas flow as well.

Reply:
Sure. Can you give me a pointer to where and I can add that to this PR too!

Reply:
https://github.com/osmlab/atlas/blob/8e2c26e3bf1621d78de19cb8612e72303f260071/src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasPointAndLineSlicer.java#L211